x/poolmanager/router_test.go

```diff
@@ -2239,12 +2630,18 @@ func (s *KeeperTestSuite) TestSplitRouteExactAmountOut() {
 			// As a result, we roughly expect the amount in to be the same
 			// as the amount out given of another token. However, the actual
 			// amount must be stricly greater than the given due to price impact.
+			multiplicativeTolerance := sdk.OneDec()
+			if tc.checkExactOutput {
+				// We set to a small value instead of zero since zero is a special case
+				// where multiplicative tolerance is skipped/not considered
+				multiplicativeTolerance = sdk.NewDecWithPrec(1, 8)
+			}
 			errTolerance := osmomath.ErrTolerance{
 				RoundingDir:             osmomath.RoundUp,
-				MultiplicativeTolerance: sdk.NewDec(1),
+				MultiplicativeTolerance: multiplicativeTolerance,
 			}
 
-			s.Require().Equal(0, errTolerance.Compare(tc.expectedTokenOutEstimate, tokenOut), fmt.Sprintf("expected %s, got %s", tc.expectedTokenOutEstimate, tokenOut))
+			s.Require().Equal(0, errTolerance.Compare(tc.expectedTokenInEstimate, tokenIn), fmt.Sprintf("expected %s, got %s", tc.expectedTokenInEstimate, tokenIn))
```

Comment:
Note that I was unable to add balance checks here due to time constraints, but the balance-related behavior does not vary between exactIn and exactOut as it operates directly on the accrued fees in `chargeTakerFee` so the coverage from this addition would have been marginal anyway